### PR TITLE
[Date Picker] fix: moment date conversion for Date Picker in native

### PR DESCRIPTION
**Pull request by Prince-Mendiratta**
### Details

The main problem that we are trying to solve in this issue is that for the date picker in Date of Birth under the Personal Details component, the selected date is not acknowledged and shows the error Date Invalid even when the date is valid in native devices on iOS and android. Moreover, the possible dates shown to the user are not consistent across platforms.

### Fixed Issues

$ https://github.com/Expensify/App/issues/14936   
PROPOSAL: https://github.com/Expensify/App/issues/14936#issuecomment-1422423006

### Tests

1. Login to newDot
2. Click on settings.
3. Click on Profile > Personal Details > Date of birth.
4. Open the date picker.
5. Click on save without entering any date. **Ensure that an error is shown**, namely `Please enter a valid date`.
6. Enter a date in the future. **Ensure that an error is shown**, namely `Date should be before 20xx-xx-xx.`
7. Enter a date before the date shown in the error above. **Ensure that the date is entered correctly and no errors are shown.**
8. After saving the date, **Ensure that the saved date is shown** in the Date of Birth section in the Personal Details.

- [x] Verify that no errors appear in the JS console

### Offline tests

N/A

### QA Steps

1. Login to newDot
2. Click on settings.
3. Click on Profile > Personal Details > Date of birth.
4. Open the date picker.
5. Click on save without entering any date. **Ensure that an error is shown**, namely `Please enter a valid date`.
6. Enter a date in the future. **Ensure that an error is shown**, namely `Date should be before 20xx-xx-xx.`
7. Enter a date before the date shown in the error above. **Ensure that the date is entered correctly and no errors are shown.**
8. After saving the date, **Ensure that the saved date is shown** in the Date of Birth section in the Personal Details.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web - Chrome</summary>

https://user-images.githubusercontent.com/54077356/218043876-1660349e-1eb5-4b3c-9a1a-165d4f6dff79.mp4

</details>

<details>
<summary>Web - Safari</summary>

https://user-images.githubusercontent.com/54077356/218044083-5956c118-aeb9-4c40-b8dd-13728b6a6025.mp4

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/54077356/218044190-3d71ef2e-f0b1-4474-bd3c-3c84717c219c.mp4

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/54077356/218044212-06c9acd5-6b8f-43d4-8b80-2a9f52075d46.mp4

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/54077356/218044282-5d834adb-6fa8-4559-be30-da9e1cd5cbc8.mp4

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/54077356/218044303-e1bd83db-5579-42c8-b714-9d55dfc1be2e.mp4

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/54077356/218044327-3693ede4-6943-49c4-ac4f-e998fb3327ac.mp4

</details>

### CODE REVIEW

File: src/pages/ReimbursementAccount/RequestorStep.js [hunk 1]
```diff
@@ -93,7 +92,7 @@ class RequestorStep extends React.Component {
         const payload = {
             bankAccountID: lodashGet(this.props.reimbursementAccount, 'achData.bankAccountID') || 0,
             ...values,
-            dob: moment(values.dob).format(CONST.DATE.MOMENT_FORMAT_STRING),
+            dob: values.dob,
```
Comment (Santhosh-Sellavel) [inline src/pages/ReimbursementAccount/RequestorStep.js:95]: same here

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]

**Comment by Santhosh-Sellavel** [comment]
Screenshot 2023-02-13 at 10 04 24 PM

**Comment by Beamanator** [comment]
@Prince-Mendiratta will you please update the PR title to be more useful? It's best if it describes exactly what you're doing in this PR

**Comment by Santhosh-Sellavel** [comment]
Android calendar is not persisted for previously selected months if before 1900. Check month is shown as January 1900

https://user-images.githubusercontent.com/85645967/218589525-78b169b8-6d4b-4976-b3bd-a8977a0c071c.mp4

**Comment by Santhosh-Sellavel** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/85645967/218589814-8fb78942-5bcf-4c4c-86fc-428cb3aa9ee8.mp4

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/85645967/218590089-cf509231-7475-4b43-a04b-5af34561f740.mp4

</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by Santhosh-Sellavel** [comment]
Will continue tomorrow

**Comment by Santhosh-Sellavel** [comment]
@Prince-Mendiratta 

Any thoughts on this https://github.com/Expensify/App/pull/15033#issuecomment-1428790410?

**Comment by Beamanator** [comment]
@Santhosh-Sellavel 

> Android calendar is not persisted for previously selected months if before 1900. Check month is shown as January 1900

Interesting find - I think this is fine for now, since we're actually building our own Date Picker at the moment, so this shouldn't be a problem in the near future 👍

**Comment by Santhosh-Sellavel** [comment]
@Beamanator 
This is not from this PR should we do something about it though?
Steps
1. Select an invalid to submit error shown
2. Exit the DOB page
3. Open again DOB page again

Values should not be persisted
https://user-images.githubusercontent.com/85645967/219462376-170b2ae2-7d14-4068-9d1a-025b54eda94e.mov

cc: @Prince-Mendiratta

**Comment by Santhosh-Sellavel** [comment]
Okay Checked other forms seems this is common to all not specific to this one.

**Review by Santhosh-Sellavel — APPROVED** [review]
LGTM tests well!

All you @Beamanator

**Comment by Santhosh-Sellavel** [comment]
If you’ve noticed earlier it's always to good bring forward for discussion. This helps speeds things!

**Review by Beamanator — APPROVED** [review]
Looks great! 👍

**Pull request merged**